pallets/aggregate/src/migrations/v2.rs

```diff
@@ -0,0 +1,214 @@
+//! Migrate from storage v1 to v2.
+
+mod v1;
+
+use frame_support::{migrations::VersionedMigration, traits::UncheckedOnRuntimeUpgrade};
+use sp_core::Get;
+use sp_std::vec::Vec;
+
+/// Implements [`UncheckedOnRuntimeUpgrade`], migrating the state of this pallet from V0 to V1.
+///
+/// In V0 of the template, the value of the [`crate::Domains`] `StorageMap` is the old `Domain`
+/// without delivery configurations and aggregation rules.
+///
+/// We migrate every domain by add a [`crate::data::AggregateSecurityRules::Untrusted`] rules
+/// and a [`crate::data::DeliveryParams`] with the same domain owner, none destination and zero
+/// price.
+///
+/// If we cannot find any delivery owner, we'll remove domain from the storage.
+pub struct InnerMigrateV1ToV2<T>(core::marker::PhantomData<T>);
+
+impl<T: crate::Config> UncheckedOnRuntimeUpgrade for InnerMigrateV1ToV2<T> {
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::TryRuntimeError> {
```

Comment:
If I remember correctly this check had some sense in the previous case where NOT all domains was migrated and so the check try to be sure that you converted all domains that is expected to convert.

But in this case maybe is a little bit use less.

Anyway we can left it here as example as well.

Reply:
yes this pre_upgrade method only keeps domains with a valid owner and I thought there was a logic I needed to keep, but anyway, yes let's keep it